public/src/components/admin/MarketState.tsx

```diff
@@ -0,0 +1,110 @@
+            this.setState(prevState => {
+                return {
+                    option: "No"
+                }
+            });
+          }catch(err){
+            console.log("Error happened while Closing Market! ", e.statusCode, e.statusMessage, e);
+            if (e.isGrpcError) {
+                showErrorNotif("Oops! Unable to reach server. Please check your internet connection!");
+            } else {
+                showErrorNotif("Oops! Something went wrong! " + e.statusMessage);
+            }
+          }
+        }
+        else{
+          // open Market
+          const MarketRequest = new OpenMarketRequest();
+          try{
+            MarketRequest.setUpdateDayHighAndLow(option);
+            const resp = await DalalActionService.openMarket(MarketRequest, sessionMd);
+            // If any error occurs, it will be raised in DalalMessage_pb_Service
+            showNotif("Market has been successfully opened");
+            this.setState(prevState => {
+                return {
+                    option: "No"
+                }
+            });
+          }catch(err){
+            console.log("Error happened while Opening Market! ", e.statusCode, e.statusMessage, e);
+            if (e.isGrpcError) {
+                showErrorNotif("Oops! Unable to reach server. Please check your internet connection!");
+            } else {
+                showErrorNotif("Oops! Something went wrong! " + e.statusMessage);
+            }
+          }
+        }
+    }
+
+    render() {
+        let marketStatus = this.props.isMarketOpen ? "red" : "green";
+        return (
+            <React.Fragment>
+                <table id="MarketState">
+                    <tbody className="ui bottom attached tab segment active inverted">
+                    <tr>
+                      <td>
+                        <label>{this.props.isMarketOpen ? "Do you want to update Previous Day Close ?" : "Do you want to update Day`s high and low ?"}</label>
+                      </td>
+                      <td>
+                        <label className="radiolabel">
+                         <input type="radio" value="Yes" onChange={(e) => {this.handleOptionChange(e.target.value)}} checked={this.state.option === "Yes"}/>Yes
+                         <span className="radiocheckmark"></span>
+                        </label>
+                        <label className="radiolabel">
+                         <input type="radio" value="No" onChange={(e) => {this.handleOptionChange(e.target.value)}} checked={this.state.option == "No"}/>No
+                         <span className="radiocheckmark"></span>
+                        </label>
+                      </td>
+                    </tr>
+                    <tr>
+                       <input type="button" className={"ui inverted " + marketStatus + " button"} onClick={(e) => {this.handleMarketState(e)}} value={this.props.isMarketOpen ? "Close Market" : "Open Market"}/>
```

Comment:
Make two separate buttons for Close Market and Open Market and update the boolean variable based on which button is clicked.